### PR TITLE
Test forms

### PR DESCRIPTION
**Pull request by siasktv**
# Description

Created e2e tests for forms routes. 

## Issue link

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature updates / changes
- [ X] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

yarn test:e2e forms

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

**Comment by cherylli** [comment]
Hi @siasktv is this ready for review? If not, please make it a draft.

**Review by JoshuaHinman — DISMISSED** [review]
All tests passed

**Comment by cherylli** [comment]
@siasktv could you please update the change log. Please make sure everything is checked in the checklist (if applicable)

**Review by JoshuaHinman — APPROVED** [review]
Ok - All tests passed

**Review by Piero7210 — APPROVED** [review]

**Comment by siasktv** [comment]
> @siasktv could you please update the change log. Please make sure everything is checked in the checklist (if applicable)

Thx! Change log has been updated as well

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,15 +19,17 @@ Another example [here](https://co-pilot.dev/changelog)
 - Add CHANGELOG.md ([#84](https://github.com/chingu-x/chingu-dashboard-be/pull/84))
 - Add Role/Permission guard ([#97](https://github.com/chingu-x/chingu-dashboard-be/pull/97))
 - Add e2e tests for auth controller ([#102](https://github.com/chingu-x/chingu-dashboard-be/pull/102))
+- Add e2e tests for techs controller ([#103](https://github.com/chingu-x/chingu-dashboard-be/pull/103))
```
Comment (cherylli) [inline CHANGELOG.md:22]: change #103 is duplicated
could you put your change in the right order please, i.e. most recent change in the bottom

**Review by cherylli — CHANGES_REQUESTED** [review]

File: test/forms.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        app.useGlobalPipes(new ValidationPipe());
+        app.use(cookieParser());
+        await app.init();
+    });
+
+    afterAll(async () => {
+        await app.close();
+    });
+
+    describe("Get Forms Endpoint Testing", () => {
```
Comment (cherylli) [inline test/forms.e2e-spec.ts:53]: could you have one "describe" for each endpoint? You can look at auth or techs e2e tests for example

Comment (siasktv) [reply test/forms.e2e-spec.ts:53]: sure no problem!

Comment (siasktv) [reply test/forms.e2e-spec.ts]: I have added one more 'describe' block for handling GET requests to retrieve a specific form (GET /forms/:formId). So, in the test suite, we now have one for handling GET requests to retrieve all forms (GET ALL /forms) and another for handling GET requests for a specific form (GET /forms/:formId), including a case to handle invalid form IDs. I don't see the need to have separate blocks for each endpoint, as /forms/${invalidFormId} is also part of retrieving a specific form. Let me know what you think :)

Comment (cherylli) [reply test/forms.e2e-spec.ts]: yes, for your case, there's no need to have nested describe

File: test/forms.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                "password",
+                app,
+            );
+            await request(app.getHttpServer())
+                .get("/forms")
+                .set("Cookie", [access_token, refresh_token])
+                .expect(200);
+        });
+    });
+    describe("GET a specific form /forms/:formId and handle a form with invalid ID", () => {
```
[the remark this answers is not in the document]
Comment (cherylli) [reply test/forms.e2e-spec.ts:66]: could you also add an asert for number of forms in the database === length of the response array

Also that /forms/:formId is returning the right form?

**Review by cherylli — CHANGES_REQUESTED** [review]

**Review by cherylli — APPROVED** [review]
thanks! All tests passed locally

**Review by JoshuaHinman — APPROVED** [review]
Tests passed

**Pull request merged**